app/services/wallets/validate_service.rb

```diff
@@ -49,7 +49,7 @@ def valid_granted_credits_amount?
     def valid_expiration_at?
       return true if args[:expiration_at].blank?
 
-      if Utils::DatetimeService.valid_format?(args[:expiration_at]) && expiration_at.to_date > Time.current.to_date
+      if Utils::Datetime.valid_format?(args[:expiration_at]) && expiration_at.to_date > Time.current.to_date
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`. (https://rubystyle.guide#if-as-a-modifier)